applications/datastore.go

```diff
@@ -0,0 +1,166 @@
+package applications
+
+import (
+	"fmt"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq" // postgres driver
+	"github.com/pkg/errors"
+	"time"
+)
+
+// Datastore manages devices in a database
+type Datastore interface {
+	NewDeviceApp(da *DeviceApplication) error
+	Applications(params ...interface{}) ([]Application, error)
+	GetApplicationsByDeviceUUID(deviceUUID string) ([]Application, error)
+	SaveApplicationByDeviceUUID(deviceUUID string, app *Application) error
+	DeleteDeviceApplications(deviceUUID string) error
+}
+
+type pgStore struct {
+	*sqlx.DB
+	logger kitlog.Logger
+}
+
+func NewDatastore(connection *sqlx.DB, logger kitlog.Logger) (Datastore, error) {
```

Comment:
What's the difference between `NewDB` and `NewDatastore`?
Why are both needed? 

At the very least add a comment on top of each to add an explanation.